### ISSUE TITLE
MavenResolverIO assumes dep_graph directory is present and crashes if it isn't

### ISSUE THREAD

**Issue by MagielBruntink** [fasten-project/fasten#464]
I'm assuming this is related to recent changes to the REST API / Maven Resolver implementation there.

Repo: with DC:

- update FASTEN REST API image to 0.0.8 in .env
- clean `docker-volumes` directory
- start the pipeline including the REST API, eg `docker-compose --profile java up -d`
- observe log of `fasten-docker-deployment_fasten-restapi-java_1`

```
2022-05-30 13:31:44.030  INFO 1 --- [nio-8080-exec-1] e.f.c.maven.resolution.MavenResolverIO   : Saving poms to /mnt/fasten/dep_graph/poms.json ...
2022-05-30 13:31:44.035 ERROR 1 --- [nio-8080-exec-1] e.f.a.restapiplugin.api.ResolutionApi    : Error constructing dependency graph maven resolver

java.lang.RuntimeException: java.io.FileNotFoundException: /mnt/fasten/dep_graph/poms.json-tmp (No such file or directory)
at eu.fasten.core.maven.resolution.MavenResolverIO.saveToDisk(Unknown Source) ~[core-0.0.8.jar!/:na]
at eu.fasten.core.maven.resolution.MavenResolverIO.loadResolver(Unknown Source) ~[core-0.0.8.jar!/:na]
```

**Comment by MagielBruntink** [comment]
It doesn't crash and seems to work if you create the `dep_graph` directory manually first.

**Comment by mir-am** [comment]
I also experienced this issue. It can be solved by manually creating the `dep_graph` directory.

**Comment by mir-am** [comment]
The issue is addressed in #467.
